evergreen/setup_baas_host.sh

```diff
@@ -124,27 +225,69 @@ fi
 
 sudo chmod 600 "${HOME}/.ssh"/*
 
+# Should an alternate data directory location be used? If so, don't init the data device
+if [[ -z "${OPT_DATA_DIR}" ]]; then
+    DATA_DIR=/data
+    init_data_device
+else
+    DATA_DIR="${OPT_DATA_DIR}"
+fi
+
+DATA_TEMP_DIR="${DATA_DIR}/tmp"
+BAAS_REMOTE_DIR="${DATA_DIR}/baas-remote"
+BAAS_WORK_DIR="${BAAS_REMOTE_DIR}/baas-work-dir"
+SERVER_PID_FILE="${BAAS_REMOTE_DIR}/baas-server.pid"
+BAAS_STOPPED_FILE="${BAAS_WORK_DIR}/baas_stopped"
+
+BAAS_PROXY_DIR="${BAAS_REMOTE_DIR}/baas-proxy-dir"
+PROXY_PID_FILE="${BAAS_REMOTE_DIR}/baas-proxy.pid"
+PROXY_STOPPED_FILE="${BAAS_PROXY_DIR}/baas_proxy_stopped"
```

Comment:
IIUC these need to match the directories used in the evg config. should we provide them as args to avoid risks of them going out of sync?

Reply:
It is tricky to keep all the directories in sync across the different scripts - I can look to make these args in a separate PR.